ir/state.cpp

```diff
@@ -255,7 +255,11 @@ State::State(const Function &f, bool source)
   : f(f), source(source), memory(*this),
     fp_rounding_mode(expr::mkVar("fp_rounding_mode", 3)),
     fp_denormal_mode(expr::mkVar("fp_denormal_mode", 2)),
-    return_val(DisjointExpr(f.getType().getDummyValue(false))) {}
+    return_val(DisjointExpr(f.getType().getDummyValue(false))) {
+
+  if (get_uf_float())
+    doesApproximation("uf-float");
```

Comment:
I am not sure what the best way to handle `doesApproximation` is with this approach. One option would be to call it in `fm_poison` in `instr.cpp`, however we don't have direct access to the UFs anymore. From what I can tell from the source code, the second argument to `doesApproximation` needs to be the APP node of the UF. However we do not have access to that anymore at this point. It may be possible to obtain the UFs by traversing the AST though. Another option might be to register a callback function with `expr.h` which is called for each created UF.